### ISSUE TITLE
Is it possible to integrate Qwt plot and qnanopainter ?

### ISSUE THREAD

**Issue by stephenlang84** [QUItCoding/qnanopainter#66]
Hi.

I'm using [Qwt lib](https://qwt.sourceforge.io/) to render a graph, my app uses qml, but as you know Qwt bases on QWidget, so I used QQuickPaintedItem to paint the output of Qwt to my qml app, inspired are from https://github.com/carlos22/qwtquick2.

But this is a performance issue [QTBUG-96205](https://bugreports.qt.io/browse/QTBUG-96205?focusedCommentId=582533&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-582533) of using QQuickPaintedItem to render a "QWidget's content".

Even though I enabled the QwtPlotOpenGLCanvas of Qwt but I found that this optimization only works for QWidget applications because [QOpenGLWidget](https://doc.qt.io/qt-6/qopenglwidget.html) does not support the render() function with OpenGL. QwtPlotOpenGLCanvas is working well with a widget application, this is example app of Qwt. https://github.com/opencor/qwt/tree/master/examples/refreshtest. 

So qnanopainter can help solving the QQuickPaintedItem's performance issue in this case. can't it ? 

Actually I tried below solution, the frequency of calling this paint() or update the graph is 40 ms, but my implementation could not solve the CPU% that takes a lot of memory, I guess it is from creating the images.
```
void QwtQuickPainter::paint(QNanoPainter *painter)
{
    static int index = 0;
    QNanoImage nImage(plotCanvas->grab(QRect(0, 0, width(), height())).toImage(), QString::number(index));
    painter->drawImage(nImage, 0, 0, width(), height());
    index++;
}
```

Thanks for your advice !

**Comment by QUItCoding** [comment]
I don't think you would get notable performance improvement with the above code. Generating images is probably the slow part and not blitting them with QNanoPainter. To get the benefits, Qwt itself would need to be ported QPainter -> QNanoPainter, which is probably quite a big task.
